internal/providers/pluginfw/tfschema/single_item_list_validator.go

```diff
@@ -0,0 +1,47 @@
+package tfschema
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+// MaxItemsValidator is a custom validator that enforces the maximum number of items in a list.
```

Comment:
Nothing against your implementation, but can we use the existing sizeAtMost validator? https://github.com/hashicorp/terraform-plugin-framework-validators/blob/main/listvalidator/size_at_most.go

This just minimizes the amount of logic that we're responsible for maintaining/testing. I think your implementation is identical but it should be OK to depend on these.